core/src/main/java/org/apache/druid/math/expr/ExprTypeConversion.java

```diff
@@ -20,24 +20,29 @@
 package org.apache.druid.math.expr;
 
 import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.Pair;
 
 import javax.annotation.Nullable;
 import java.util.List;
 
 public class ExprTypeConversion
 {
+  public static Pair<ExprType, ExprType> coerceNull(Expr.InputBindingInspector inspector, Expr left, Expr right)
```

Comment:
i did one better and just removed it. null literals output type is now `null` instead of their type, so that they behave and are handled like missing columns (since null doesn't have a type)